### PR TITLE
Fix some videos with uppercase file extensions not displaying correctly in-game

### LINKED ISSUE #23259: Importing certain beatmaps doesn't import background image and/or video
### Type

Game behaviour

### Bug description

Importing certain beatmaps like https://osu.ppy.sh/beatmapsets/15639 doesn't import background images and videos. This is a regression from 2023.403.1.

### Screenshots or videos

osu_2023-04-21_01-24-26

https://user-images.githubusercontent.com/46954095/233507359-a8a8ea6c-9aae-470d-a029-0928f10767a8.mp4

### Version

2023.419.0

### Logs

[performance.log](https://github.com/ppy/osu/files/11291082/performance.log)
[runtime.log](https://github.com/ppy/osu/files/11291083/runtime.log)
[updater.log](https://github.com/ppy/osu/files/11291084/updater.log)
[database.log](https://github.com/ppy/osu/files/11291085/database.log)
[input.log](https://github.com/ppy/osu/files/11291086/input.log)

client.realm after import: [client.realm.zip](https://github.com/ppy/osu/files/11291090/client.realm.zip)

### PR DESCRIPTION
**Pull request by Haspamelodica**
This PR fixes #23259.

The reason for this bug is that the file extension of the video is checked against a list of accepted video file extensions. However, this comparison is case-sensitive, and the list of accepted extensions is all-lowercase. So if a beatmap has a background video with an uppercase file extension, like the beatmap mentioned in the original bug report (https://osu.ppy.sh/beatmapsets/15639: `Video.AVI`), it is interpreted as an image file incorrectly specified as a video (see da947d86613ffe99f19f4d49a6535810ba752c59 for details about that), which causes both the video and original background image to break.

This PR fixes this bug by changing the list to all-uppercase, and converting all extensions to uppercase (using `ToUpperInvariant`) before checking them against the list. I used uppercase instead of lowercase because of https://learn.microsoft.com/de-de/previous-versions/visualstudio/visual-studio-2015/code-quality/ca1308-normalize-strings-to-uppercase?view=vs-2015&redirectedfrom=MSDN (this is further explaind in https://stackoverflow.com/questions/2801508/what-is-wrong-with-tolowerinvariant and https://stackoverflow.com/questions/6225808/string-tolower-and-string-tolowerinvariant).

### CODE REVIEW

File: osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs [hunk 1]
```diff
@@ -369,7 +369,7 @@ private void handleEvent(string line)
                     // Some very old beatmaps had incorrect type specifications for their backgrounds (ie. using 1 for VIDEO
                     // instead of 0 for BACKGROUND). To handle this gracefully, check the file extension against known supported
                     // video extensions and handle similar to a background if it doesn't match.
-                    if (!OsuGameBase.VIDEO_EXTENSIONS.Contains(Path.GetExtension(filename)))
+                    if (!OsuGameBase.VIDEO_EXTENSIONS.Contains(Path.GetExtension(filename).ToUpperInvariant()))
```
Comment (peppy) [inline osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs:372]: We're using `ToLower` elsewhere, so please stick to using this.

https://github.com/ppy/osu/blob/b8904fe7474e17bf48a7dbbbd60ae338dd77529c/osu.Game/OsuGameBase_Importing.cs#L35

File: osu.Game/OsuGameBase.cs [hunk 2]
```diff
@@ -71,7 +71,7 @@ namespace osu.Game
     [Cached(typeof(OsuGameBase))]
     public partial class OsuGameBase : Framework.Game, ICanAcceptFiles, IBeatSyncProvider
     {
-        public static readonly string[] VIDEO_EXTENSIONS = { ".mp4", ".mov", ".avi", ".flv", ".mpg", ".wmv", ".m4v" };
+        public static readonly string[] VIDEO_EXTENSIONS = { ".MP4", ".MOV", ".AVI", ".FLV", ".MPG", ".WMV", ".M4V" };
```
Comment (peppy) [inline osu.Game/OsuGameBase.cs:74]: Revert this change please.

**Review by peppy — CHANGES_REQUESTED** [review]

File: osu.Game/OsuGameBase.cs (hunk 2 above)
Comment (Haspamelodica) [reply osu.Game/OsuGameBase.cs]: Done

File: osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs (hunk 1 above)
Comment (Haspamelodica) [reply osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs]: Done, switched to lowercase now.

**Comment by peppy** [comment]
I'd probably feel a lot safer having test coverage added for this. It should be pretty easy to do so (see the PR you linked which regressed this behaviour in the first place – it has a very similar test setup).

**Comment by Haspamelodica** [comment]
Added test cases now. Using the opportunity I also a misleading

**Comment by peppy** [comment]
The tests seems to be passing without your changes, can you double check they are testing the case that was broken?

**Comment by Haspamelodica** [comment]
That's weird; at least for me the tests fail without my changes. 
grafik

**Comment by Haspamelodica** [comment]
I just double-checked whether the tests actually test the broken case, and it really seems to me that they do. @peppy Could you give more information on the circumstances under which the tests succeed even without my changes?

grafik
grafik

These are the exact patches I used to double-check: (applied to the current commit on master, 36113f60b7d1cbcce7870d579a61cbdc9d4384c8)
[fix_only.patch](https://github.com/ppy/osu/files/11302100/fix_only.patch)
[tests_and_fix.patch](https://github.com/ppy/osu/files/11302101/tests_and_fix.patch)
[tests_only.patch](https://github.com/ppy/osu/files/11302102/tests_only.patch)

**Comment by Haspamelodica** [comment]
Is there something you're waiting for me to do or am I too impatient?

**Comment by peppy** [comment]
The latter.

**Review by peppy — APPROVED** [review]
